server/chat.ts

```diff
@@ -1293,509 +1088,763 @@ export class CommandContext extends MessageContext {
 	}
 }
 
-/**
- * Command parser
- *
- * Usage:
- *   Chat.parse(message, room, user, connection)
- *
- * Parses the message. If it's a command, the command is executed, if
- * not, it's displayed directly in the room.
- *
- * Examples:
- *   Chat.parse("/join lobby", room, user, connection)
- *     will make the user join the lobby.
- *
- *   Chat.parse("Hi, guys!", room, user, connection)
- *     will return "Hi, guys!" if the user isn't muted, or
- *     if he's muted, will warn him that he's muted.
- *
- * The return value is the return value of the command handler, if any,
- * or the message, if there wasn't a command. This value could be a success
- * or failure (few commands report these) or a Promise for when the command
- * is done executing, if it's not currently done.
- *
- * @param message - the message the user is trying to say
- * @param room - the room the user is trying to say it in
- * @param user - the user that sent the message
- * @param connection - the connection the user sent the message from
- */
-function parse(message: string, room: Room, user: User, connection: Connection) {
-	Chat.loadPlugins();
-	const context = new CommandContext({message, room, user, connection});
+export const Chat = new class {
+	constructor() {
+		this.multiLinePattern = new PatternTester();
```

Comment:
I would rather define these in the declaration than in the constructor. We don't usually do it because of some minor overhead from sucrase, but for literal one-time-use `new class` classes, I'd prioritize readability.